### ISSUE TITLE
Problem: rust work will go unused in racket implementation

### ISSUE THREAD

**Issue by sjmackenzie** [fractalide/fractalide#93]
Potential solution:

Use racket for the gui + editor + graph manipulation layers then when at component level one could choose to implement a node in racket but if one needs performance it should be simple to reuse rustfbp and ffi into it at node level. [0]

Nix infra for racket needs to be extended to support building rust code too.

A further question: Can the rust scheduler be reused?

 [0] https://docs.racket-lang.org/foreign/index.html

**Comment by clacke** [comment]
https://docs.racket-lang.org/inside/embedding.html

**Comment by clacke** [comment]
In the very long term, #lang rust might be relevant: https://www.mail-archive.com/racket-users@googlegroups.com/msg37330.html

**Comment by sjmackenzie** [comment]
Yeah I've been following the `#lang racket` thread it's just idea stage but even if it was completed it wouldn't be as fast as Rust. One major reason for using racket is that it has tight interaction with the GUI. In this way we can still leverage the speed of rust by embedding it in a racket node, which still allows gui manipulation if done between the time of when data enters a racket node and is sent on into an embedded rust node. So the best of both worlds is possible. Those rust nodes could just be plain old rust nodes sitting in the `nodes/rs/*` directory and still be composed into other pure rust hierarchies.

**Comment by clacke** [comment]
You mean use the sub-fbp rust as a C library and have racket code talk fbp to other racket code?

**Comment by sjmackenzie** [comment]
Yes, `data-in -> racket_fbp( <potential-racket-gui-effects> -> rust_fbp( -> transform -> ) -> <potential-racket-gui-effects>) -> ) -> data-out`
